fix raxx.new failing on Elixir 1.9

edbedbe opened the pull request:
Thanks for this project! I've been really enjoying the simplicity of the interface. 

`raxx.new` fails on Elixir 1.9 due to what appears to be a different return value from `Mix.Generator.create_directory`.  

```
> mix raxx.new project
* creating project
** (MatchError) no match of right hand side value: true
    lib/raxx/kit.ex:18: Raxx.Kit.generate/1
    lib/mix/tasks/raxx/new.ex:64: Mix.Tasks.Raxx.New.run/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:79: Mix.CLI.run_task/2
```

It looks like it used to return `:ok`, and now it always returns `true`. To not break backwards compatibility with different Elixir versions, I didn't match with `true`, but just ignored the return value instead. After this change it works on Elixir 1.9 for me.

CrowdHailer commented:
Would you mind putting a comment to the effect that this function returns `:ok` pre 1.9 and now `true`. I think we should use a case and handle :ok or true. I want it to crash in case of false/error so that it doesn't continue in the case when creating the directory fails.

Otherwise it looks like a really helpful fix. Thanks, because i'm not getting to 1.9 for a bit.

If you have time a CHANGELOG entry would be great. I'll make a quick release and put this all under version 0.12.1

edbedbe commented:
Thanks for reviewing! So after a bit more investigation, `Mix.Generator.create_directory` has used `File.mkdir_p!` since at least [1.0](https://github.com/elixir-lang/elixir/blob/v1.0/lib/mix/lib/mix/generator.ex#L29), so it raises a `File.Error` exception when it fails instead of returning an error. I guess it would be more consistent with convention if the function was `Mix.Generator.create_directory!` because that's how it behaves. 

In [1.9](https://github.com/elixir-lang/elixir/blob/v1.9/lib/mix/lib/mix/generator.ex#L54), `Mix.Generator.create_directory` was just changed to just return `true` but it still uses `File.mkdir_p!` and throws an exception on a failure. The return values in all versions since 1.0 are not meaningful since in an error case it will never return. 

I tested this change with Elixir 1.8 and 1.9, and `raxx.new` exhibits the desired behavior and still exits early on a failure creating the directory due the the thrown exception:

```
> mix raxx.new /usr/bin/raxx_demo
* creating /usr/bin/raxx_demo
** (File.Error) could not make directory (with -p) "/usr/bin/raxx_demo": not owner
    (elixir) lib/file.ex:314: File.mkdir_p!/1
    lib/raxx/kit.ex:19: Raxx.Kit.generate/1
    lib/mix/tasks/raxx/new.ex:64: Mix.Tasks.Raxx.New.run/1
    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:79: Mix.CLI.run_task/2
```

I updated the changelog and added a comment I thought could be helpful. Let me know if i should change anything.

CrowdHailer commented:
Published as version [0.12.1](https://hex.pm/packages/raxx_kit)

If you feel like it drop a comment here https://elixirforum.com/t/raxx-kit-micro-framework-for-web-applications/13805/17

edbedbe commented:
Awesome, thank you! That was head-spinningly fast. I posted an update on that thread. I just joined so it's pending moderator approval.
